src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -71,9 +76,11 @@ class TaskBuilder(app: AppDefinition,
       .addResources(ScalarResource(Resource.CPUS, app.cpus, cpuRole))
       .addResources(ScalarResource(Resource.MEM, app.mem, memRole))
 
-    if (portsResource.ranges.nonEmpty) {
+    if (labels.nonEmpty)
```

Comment:
If its not present, we shouldn't set it. First because it will simply send more bytes over the wire, second because default values will be ignored in that case.